sdks/python/apache_beam/io/filebasedsink.py

```diff
@@ -178,85 +178,116 @@ def open_writer(self, init_result, uid):
     return FileBasedSinkWriter(self, os.path.join(init_result, uid) + suffix)
 
   @check_accessible(['file_path_prefix', 'file_name_suffix'])
-  def finalize_write(self, init_result, writer_results):
-    file_path_prefix = self.file_path_prefix.get()
-    file_name_suffix = self.file_name_suffix.get()
+  def _get_final_name(self, shard_num, num_shards):
+    return ''.join([
```

Comment:
<!--new_thread; commit:bc394b7dd981af5640e2e80ebcc83e86eee0136e; resolved:0-->
Do we change the format of the final file names here (which would be backwards incompatible) or is this just refactoring ?

Reply:
Filename format is the same (verified by existing unit tests). This was refactored to let pre_finalize() access the filenames as well.